### PR TITLE
feat(core): add sign-in-mode

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add sign-in-mode to sign-in-experience settings.

1. Introduce sign-in-mode to the sign-in-experience settings.  `SignIn`, `Register` and `SignInAndRegister`
2. Update SignInSettings API, for admin console, always set signInMethods to `username`. If no user is found set sign-in mode to Register, else set to SignIn. 
3. Update UI signin page settings to reflect the updates
image

## Linear Issue Reference

log-2870

## Testing

test locally
ut case update
@logto-io/eng

### CODE REVIEW

**Review by wangsijie — APPROVED** [review]
LGTM
So, in the future, "sign in experience" settings is a sub resource of application?

File: packages/core/src/routes/sign-in-settings.ts [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-export default function signInSettingsRoutes<T extends AnonymousRouter>(router: T) {
+export default function signInSettingsRoutes<T extends AnonymousRouter>(
+  router: T,
+  provider: Provider
+) {
   router.get(
     '/sign-in-settings',
     async (ctx, next) => {
-      const [signInExperience, connectorInstances] = await Promise.all([
+      const [signInExperience, connectorInstances, interactions] = await Promise.all([
```
Comment (IceHe) [inline packages/core/src/routes/sign-in-settings.ts:20]: should be singular: interactions → interaction

File: packages/schemas/tables/sign_in_experiences.sql [hunk 2]
```diff
@@ -1,9 +1,13 @@
+create type sign_in_mode as enum ('SignIn', 'Register', 'SignInAndRegister');
+
+
```
Comment (IceHe) [inline packages/schemas/tables/sign_in_experiences.sql:3]: just keep one blank line

**Comment by simeng-li** [comment]
> LGTM So, in the future, "sign in experience" settings is a sub resource of application?

I believe so. Eventually, we should allow application-level sign-in-exp configurations.

File: packages/schemas/tables/sign_in_experiences.sql (hunk 2 above)
Comment (simeng-li) [reply packages/schemas/tables/sign_in_experiences.sql]: updated

File: packages/core/src/routes/sign-in-settings.ts (hunk 1 above)
Comment (simeng-li) [reply packages/core/src/routes/sign-in-settings.ts]: yep updated

**Pull request merged**